runconfig/errors.go

```diff
@@ -5,8 +5,10 @@ const (
 	ErrConflictContainerNetworkAndLinks validationError = "conflicting options: container type network can't be used with links. This would result in undefined behavior"
 	// ErrConflictSharedNetwork conflict between private and other networks
 	ErrConflictSharedNetwork validationError = "container sharing network namespace with another container or host cannot be connected to any other network"
-	// ErrConflictHostNetwork conflict from being disconnected from host network or connected to host network.
-	ErrConflictHostNetwork validationError = "container cannot be disconnected from host network or connected to host network"
+	// ErrConflictConnectToHostNetwork error when attempting to connect a container to host network when not in host network mode
+	ErrConflictConnectToHostNetwork validationError = "cannot connect container to host network - container must be created in host network mode"
+	// ErrConflictDisconnectFromHostNetwork error when attempting to disconnect a container from host network when in host network mode
+	ErrConflictDisconnectFromHostNetwork validationError = "cannot disconnect container from host network - container was created in host network mode"
```

Comment:
Not a problem with this pull request, but we really need to look at removing these exported errors; these errors are not used as special (sentinel) errors, and only defined out of convenience; looks like the only place they're used outside of this package is in some tests (for string-matching), which _ideally_ would check for the errdefs type ("invalid parameter" or "conflict", depending on what applies).

I don't think this pull request makes the situation much worse though, so probably not a problem to add on/two more.


I've done a quick check to see if anyone outside our own codebase is using the `ErrConflictHostNetwork` error, but it doesn't appear to be, so removing that const shouldn't be a problem AFAICS; https://grep.app/search?q=.ErrConflictHostNetwork